pkgs/applications/editors/emacs/default.nix

```diff
@@ -9,8 +9,13 @@ lib.makeScope pkgs.newScope (self:
 
       inherit (pkgs.darwin) sigtool;
       inherit (pkgs.darwin.apple_sdk.frameworks)
-        AppKit Carbon Cocoa GSS ImageCaptureCore ImageIO IOKit OSAKit Quartz
-        QuartzCore WebKit;
+        Accelerate AppKit Carbon Cocoa GSS ImageCaptureCore ImageIO IOKit OSAKit
+        Quartz QuartzCore WebKit;
+
+      UniformTypeIdentifiers =
+        if lib.versionAtLeast pkgs.stdenv.hostPlatform.darwinMinVersion "11"
+          then pkgs.darwin.apple_sdk.frameworks.UniformTypeIdentifiers
+          else null;
```

Comment:
Get rid of `null`ities. It is deprecated and serves no purpose besides heisenbugging the code. I will save you a Google trip:

https://archive.is/5WnHy
https://archive.ph/DP923
https://www.youtube.com/watch?v=e0tstsbD4Ro

Indeed, this same Emacs derivation was plagued with `null`s:

https://github.com/NixOS/nixpkgs/pull/233301#issuecomment-1564348451

> Next, I plan to remove the `? null's`. This will require including Boolean guards like `fancylibSupport ? false, fancylib`.

https://github.com/NixOS/nixpkgs/pull/233301#issuecomment-1567661913

> I think you accidentally fixed a packaging bug where gpm wasn't enabled even though withGpm was set to true

Do something like

```
enableUniformTypeIdentifiers ? (stdenv.isDarwin && lib.versionAtLeast pkgs.stdenv.hostPlatform.darwinMinVersion "11"),
```

and use it to control the inclusion of `UniformTypeIdentifiers` in the code.